### ISSUE TITLE
Add type formatter to Log keyword

### ISSUE THREAD

**Issue by MaxRusse11** [robotframework/robotframework#4095]
Add a formatter=type to Log keyword for verification of variable type

**Comment by pekkaklarck** [comment]
I think this would be a good idea. We already support `formatter=repr` and `formatter=ascii` to get the message formatted using Python's `repr()` and `ascii()` [built-in functions](https://docs.python.org/3/library/functions.html), respectively. Supporting also `type()` would be convenient and totally backwards compatible.

We could also consider adding support for `formatter=<any-built-in>` so that you could use also e.g. `formatter=len`. I'm not sure are there many/any other built-in that could even in theory be useful as such a formatter, but implementation this way would probably be simpler than handling named formatters specifically.

We could also consider `formatter=<any-callable>` so that we'd evaluate the specified formatter in Python similarly as we evaluate expressions using the Evaluate keyword. Because our evaluation logic imports modules automatically, this would allow thinks like `formatter=math.ceil`. This gets more complicated and is also harder to explain than just being able to use any built-in. Probably not worth the effort (and can be implemented later if needed).

Finally, we could consider supporting multiple formatters like `formatter=str,type` to get the message logged multiple types. The output
```
Log    example    formatter=str,repr,type
```
could be something like this:
```
str:    example
repr:   'example'
type:   string
```

**Comment by pekkaklarck** [comment]
This was implemented in PR #4126 by @Wolfe1. Thanks for contribution!

The PR added both `type` and `len` formatters. We agreed there aren't other Python builtins that could really be used here and that supporting any Python function is probably not worth the effort. We could still consider adding support for multiple formatters (e.g. `formatter=type,repr`) like I proposed in a comment above, but a separate issue can be submitted about that.

**Comment by pekkaklarck** [comment]
I realized it's better to log only the name of the type, not the type itself, with the `type` formatter. That changes, for example, `<class 'str'>` to just `str`. I'll do that change myself. I'll also add a note that `type` and `len` are new in RF 5.0 to the documentation.
